src/AuthEncoding/AuthEncoding.py

```diff
@@ -162,6 +162,43 @@ def validate(self, reference, attempt):
 registerScheme(u'SHA256', SHA256DigestScheme())
 
 
+# Bcrypt support may not have been requested at installation time
+# - installed via the 'bcrypt' extra
+try:
+    import bcrypt
+except ImportError:
+    bcrypt = None
+
+
+if bcrypt is not None:
+
+    # bcrypt routines require input as bytes.
+
+    def _ensure_bytes(pw, encoding='utf-8'):
+        """Ensures the given password `pw` is returned as bytes.
+        """
+        if isinstance(pw, six.text_type):
+            pw = pw.encode(encoding)
+        return pw
+
+    class BCRYPTHashingScheme:
+         """A BCRYPT hashing scheme."""
+
+         def encrypt(self, pw):
+             return bcrypt.hashpw(_ensure_bytes(pw), bcrypt.gensalt())
+
+         def validate(self, reference, attempt):
+             try:
+                 hashed = bcrypt.hashpw(_ensure_bytes(attempt), reference)
+             except ValueError:
+                 valid = False
+             else:
+                 valid = hashed == reference
```

Comment:
Yes, this was a very dumb attempt/implementation from the original code, I seem to recall "fixing" this in previous code somewhere in the distance past... *sighs*   - fixed in  db76353071f28fa1ff02b7a02b602bd6fe418a18
FWIW, there's a great deal of overlap/duplication of this AuthEncoding API vs zope.password,
where I'd previously put work in to [attempt](https://github.com/zopefoundation/zope.password/pull/1) to implement `bcrypt` support.